src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -0,0 +1,122 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { initializeConnector, Web3ReactHooks, Web3ReactProvider } from '@web3-react/core'
+import { EIP1193 } from '@web3-react/eip1193'
+import { MetaMask } from '@web3-react/metamask'
+import { Network } from '@web3-react/network'
+import { Connector, Provider as Eip1193Provider, Web3ReactStore } from '@web3-react/types'
+import { Url } from '@web3-react/url'
+import { WalletConnect } from '@web3-react/walletconnect'
+import { Buffer } from 'buffer'
+import { SupportedChainId } from 'constants/chains'
+import { PropsWithChildren, useMemo } from 'react'
+import { useCallback } from 'react'
+
+export let connections: [Connector, Web3ReactHooks][] = []
+export type Web3Connection = [Connector, Web3ReactHooks]
+
+function toWeb3Connection<T extends Connector>([connector, hooks]: [T, Web3ReactHooks, Web3ReactStore]): [
+  T,
+  Web3ReactHooks
+] {
+  return [connector, hooks]
+}
+
+function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
+  if (!provider) return
+  if (JsonRpcProvider.isProvider(provider)) {
+    return toWeb3Connection(initializeConnector((actions) => new Url({ actions, url: provider })))
+  } else if (JsonRpcProvider.isProvider((provider as any).provider)) {
+    throw new Error('Eip1193Bridge is experimental: pass your ethers Provider directly')
+  } else {
+    return toWeb3Connection(initializeConnector((actions) => new EIP1193({ actions, provider })))
+  }
+}
+
+function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
+  // TODO(kristiehuang): implement RPC URL fallback, then jsonRpcEndpoint can be optional
+
+  // WalletConnect relies on Buffer, so it must be polyfilled.
+  if (!('Buffer' in window)) {
```

Comment:
can you polyfill this outside of this function?

Reply:
where would this typically go?

Reply:
usually in `polyfills.ts` file which is imported in `index.tsx`